apps/dashboard/src/components/activity/activity-table.tsx

```diff
@@ -0,0 +1,171 @@
+              onMouseLeave={handleRowMouseLeave}
+            >
+              <TableCell className="px-3">
+                <div className="flex flex-col">
+                  <span className="text-foreground-950 font-medium">
+                    {activity.template?.name || 'Deleted workflow'}
+                  </span>
+                  <span className="text-foreground-400 text-[10px] leading-[14px]">
+                    {activity.transactionId} {getSubscriberDisplay(activity.subscriber)}
+                  </span>
+                </div>
+              </TableCell>
+              <TableCell className="px-3">
+                <StatusBadge jobs={activity.jobs} />
+              </TableCell>
+              <TableCell className="px-3">
+                <StepIndicators jobs={activity.jobs} />
+              </TableCell>
+              <TableCell className="text-foreground-600 px-3">
+                <TimeDisplayHoverCard date={new Date(activity.createdAt)}>
+                  <span>{formatDate(activity.createdAt)}</span>
+                </TimeDisplayHoverCard>
+              </TableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+
+      <Pagination offset={offset} limit={limit} hasMore={hasMore} onOffsetChange={handleOffsetChange} />
+    </div>
+  );
+}
+
+function formatDate(date: string) {
+  return format(new Date(date), 'MMM d yyyy, HH:mm:ss');
+}
+
+function SkeletonRow() {
+  return (
+    <TableRow className="animate-pulse">
+      <TableCell className="px-3">
+        <div className="flex flex-col gap-1">
+          <div className="h-5 w-36 rounded bg-neutral-200" />
+          <div className="h-2.5 w-20 rounded bg-neutral-100" />
+        </div>
+      </TableCell>
+      <TableCell className="px-3">
+        <div className="flex h-7 w-28 items-center justify-center gap-1.5 rounded-full bg-neutral-100">
+          <div className="h-3.5 w-3.5 rounded-full bg-neutral-200" />
+          <div className="h-3.5 w-16 rounded bg-neutral-200" />
+        </div>
+      </TableCell>
+      <TableCell className="px-3">
+        <div className="flex items-center">
+          {[...Array(3)].map((_, i) => (
+            <div
+              key={i}
+              className="-ml-2 flex h-7 w-7 items-center justify-center rounded-full bg-neutral-100 first:ml-0"
+            >
+              <div className="h-4 w-4 rounded bg-neutral-200" />
+            </div>
+          ))}
+        </div>
+      </TableCell>
+      <TableCell className="px-3">
+        <div className="h-4 w-36 rounded bg-neutral-100 font-mono" />
+      </TableCell>
+    </TableRow>
+  );
+}
+
+function getSubscriberDisplay(subscriber?: Pick<ISubscriber, '_id' | 'subscriberId' | 'firstName' | 'lastName'>) {
+  if (!subscriber) return '';
+
+  if (subscriber.firstName || subscriber.lastName) {
+    return `• ${subscriber.firstName || ''} ${subscriber.lastName || ''}`;
+  }
+
+  return '';
```

Comment:
Shall we fallback to email and phone or ID? It would look better than an empty column.

Reply:
I was considering this, let's add it. Why not :)